.github/workflows/pre-commit.yml

```diff
@@ -14,7 +14,7 @@ jobs:
     env:
       # these checks currently require a ridiculous number of dependencies :)
       # consider repackaging them in a container in the future
-      SKIP: "web-js-lint-check,web-css-lint-check,web-misc-lint-check,black"
+      SKIP: "web-js-lint-check,web-css-lint-check,web-misc-lint-check,black,golangci-lint"
```

Comment:
Why are we skipping this in the github workflow?  Is it not possible to make it work consistently, perhaps with the script update below?

Reply:
Erik probably has some comments but from my pov:
I agree it'd be good to not skip this but this is what I shared in slack w/ @erikwilson 
> not erroring in ci or even locally if the binary isn't found sounds like a net positive when compared to not having the hook at all so I'm for it. these checks are enforced by ci elsewhere anyway

plus: who is using it (there is a poll in the BE team's channel, timing, and priorities, and that we want to address [an issue ](https://hpe-aiatscale.slack.com/archives/C04FMDK85U3/p1688059675997919?thread_ts=1688045294.192889&cid=C04FMDK85U3) asap

Reply:
It was not actually working in the first place, we were just silently skipping if the command was not installed, here it is just explicitly being skipped and we now error if the command does not exist.

We are calling `make check` in CircleCI anyhow, which is the preferred way of linting.

Reply:
Ok, so if `make check` is doing everything we want this to check to do and this check is being skipped both locally and (soon) in github, would it perhaps make more sense to either remove this entirely, or have pre-commit just run `make check`?

The point of pre-commit is to shift the lighter-weight checks left so developers can essentially fail faster.  Is this change helping identify future CI failures faster?  If not, I'd say to scrap it. :)

Reply:
pre-commit is typically run locally so I am a little confused why it is being run in CI, especially since we are skipping so many hooks. 

I think it should be okay to remove the pre-commit github workflow, maybe best done as another PR.

Reply:
altho pre-commit here may help pick up changes that `make check` may not notice, like the shell formatter hook or similar things

Reply:
We run it in CI because there's no other way to enforce it being run locally.  This brings parity to checks run by developers who do and do not install pre-commit.  However, that proposition hinges upon the checks in pre-commit actually being "important" checks which should be required of everyone.  In some cases, it's easier to implement checks with pre-commit modules than with local make commands, as pre-commit can source modules in a variety of ways.  Anyone who has set up the determined build environment knows that it's enough of a nightmare already without also adding in some new obscure linting tool or whatever. :exploding_head: 

The checks which are currently skipped (prior to this PR) are web linter checks which have a pretty large set of dependencies and which are a little slow to run, so they're being skipped until we figure out a good way to bundle those dependencies without blowing the whole github budget on downloading them live every time (and without enduring the inevitable failures when there's an NPM networking hiccup or similar).  Ideally there would be no skips, but those are in there now as a time v/s value compromise.  It's been proposed that those scripts could be repackaged as docker images which can be native pre-commit modules, which would make everyone's lives easier and allow them to be un-skipped again.  But that's tech debt for another time.

Reply:
Actually, I forget exactly why "black" is being skipped.  I think there was a version mismatch which was hard to reconcile in pre-commit.  We should verify that it actually still _needs_ skipped.

Reply:
I think ideally `make check` would run pre-commit to help avoid duplication of responsibilities, and then all of the pre-commit checks could be run in CircleCI where the dependencies would be installed and all of the hooks could be run. Then we could drop this pre-commit github workflow without being worried about losing coverage.

Reply:
Whether make check runs pre-commit or vice-versa  is up for debate. I lean a little more in the direction of pre-commit being the primary orchestrator for checks which should run before every commit, but I'm open to persuasion.

As for why it's run here...  That's basically tech debt now.  It's a little easier to run pre-commit out of github actions than CircleCI because of the upstream-maintained github action, and "everything is going to Actions" was the strategy at the time.  That has changed.  Once we get the CircleCI config wrestled into a little easier to manage state (that's major work currently planned to start in the next couple of months), we probably will want to move the pre-commit execution to Circle to better align with the intended separation between repo management in actions and test execution in Circle.

Reply:
> Whether make check runs pre-commit or vice-versa is up for debate. I lean a little more in the direction of pre-commit being the primary orchestrator for checks which should run before every commit, but I'm open to persuasion.

+1 our pre-commit checks in general do a better job for the things they cover compared to our make targets AFAIK. it'd be nice if one works through the other.